### PR TITLE
bug fixes for fi_collective

### PR DESCRIPTION
**Pull request by zhenggb72**
three commits to fix three bugs in fi_collective

### CODE REVIEW

File: prov/util/src/util_coll.c [hunk 1]
```diff
@@ -773,12 +773,14 @@ static int util_coll_find_local_rank(struct fid_ep *ep, struct util_coll_mc *col
 	size_t addrlen;
 	char *addr;
 	int ret, mem;
+	fi_addr_t  my_addr;
+        int i;
```
Comment (shefty) [inline prov/util/src/util_coll.c:777]: spacing

File: prov/util/src/util_coll.c [hunk 2]
```diff
@@ -773,12 +773,14 @@ static int util_coll_find_local_rank(struct fid_ep *ep, struct util_coll_mc *col
...
 
 	addrlen = sizeof(mem);
 	addr = (char *) &mem;
 
 	ret = fi_getname(&ep->fid, addr, &addrlen);
-	if (ret != -FI_ETOOSMALL) {
+	if (ret != -FI_ETOOSMALL && sizeof(size_t) >= addrlen) {
```
Comment (shefty) [inline prov/util/src/util_coll.c:783]: The flow through here does not look correct.  We should just set addrlen = 0 and addr = NULL, get the correct size, and fall through so that lookup below can be done.

Please update the commit message with a signed-off-by line, and change the commit title to:
```
util/coll: Find correct local collective rank

File: prov/util/src/util_coll.c [hunk 3]
```diff
@@ -791,8 +793,16 @@ static int util_coll_find_local_rank(struct fid_ep *ep, struct util_coll_mc *col
...
-	coll_mc->local_rank =
-		ofi_av_lookup_fi_addr(coll_mc->av_set->av, addr);
+	my_addr = ofi_av_lookup_fi_addr(coll_mc->av_set->av, addr);
+
+	coll_mc->local_rank = FI_ADDR_NOTAVAIL;
+        if (my_addr != FI_ADDR_NOTAVAIL) {
+		for (i=0; i<coll_mc->av_set->fi_addr_count; i++)
+			if (coll_mc->av_set->fi_addr_array[i] == my_addr) {
+			coll_mc->local_rank = i;
+			break;
```
Comment (shefty) [inline prov/util/src/util_coll.c:803]: indentation is off.  spacing is also off in places.  Make sure to use tabs for indentation.

File: prov/util/src/util_coll.c [hunk 4]
```diff
@@ -1033,18 +1033,16 @@ int ofi_join_collective(struct fid_ep *ep, fi_addr_t coll_addr,
...
 
-		ret = ofi_bitmask_create(&join_op->data.join.tmp, OFI_MAX_GROUP_ID);
-		if (ret)
-			goto err3;
+	ret = ofi_bitmask_create(&join_op->data.join.tmp, OFI_MAX_GROUP_ID);
+	if (ret)
+		goto err3;
 
-	} else {
+	if (new_coll_mc->local_rank == FI_ADDR_NOTAVAIL)
```
Comment (shefty) [inline prov/util/src/util_coll.c:1044]: Please update commit message to:
```
util/coll: <Describe what is being fixed>
```
and update the message to indicate why the change is being made, because i'm not following it from the commit message.

File: prov/util/src/util_coll.c (hunk 1 above)
Comment (zhenggb72) [reply prov/util/src/util_coll.c]: fixed

File: prov/util/src/util_coll.c (hunk 3 above)
Comment (zhenggb72) [reply prov/util/src/util_coll.c]: fixed

File: prov/util/src/util_coll.c (hunk 4 above)
Comment (zhenggb72) [reply prov/util/src/util_coll.c]: fixed

**Comment by shefty** [comment]
@ooststep - please review
